core/src/test/scala/integration/kafka/server/FetchRequestTestDowngrade.scala

```diff
@@ -43,8 +43,9 @@ class FetchRequestTestDowngrade extends BaseRequestTest {
         )
     }
 
-    @Test
-    def testTopicIdIsRemovedFromFetcherWhenControllerDowngrades(): Unit = {
+    @ParameterizedTest(name = TestInfoUtils.TestWithParameterizedQuorumAndGroupProtocolNames)
+    @MethodSource(Array("getTestQuorumAndGroupProtocolParametersClassicGroupProtocolOnly_ZK_implicit"))
+    def testTopicIdIsRemovedFromFetcherWhenControllerDowngrades(quorum: String, groupProtocol: String): Unit = {
```

Comment:
This test case is for the ZK-only API (LeaderAndIsrRequest). Not sure if we plan to remove all ZK-only APIs in 4.0... cc @dajac.

Reply:
There are around 20 integration tests that are effectively Zookeeper-only. In some cases, nothing in the test explicitly calls that dependency out. I introduced the `@MethodSource` `getTestQuorumAndGroupProtocolParametersClassicGroupProtocolOnly_ZK_implicit()` which includes the test parameters `quorum=zk` and `groupProtocol=classic`. The idea is that someone more knowledgeable about the tests would review these and either a) remove the test, or b) update the test to run against Kraft, if possible.

Reply:
> The idea is that someone more knowledgeable about the tests would review these and either a) remove the test, or b) update the test to run against Kraft, if possible.

Yes, that will be a great reference, similar to #17727.

For example, FetchRequestTestDowngrade is for LeaderAndIsrRequest, which is ZK-only, so it's okay to remove it in a follow-up (https://issues.apache.org/jira/browse/KAFKA-17976).